app/components/img-fancy-load.js

```diff
@@ -0,0 +1,37 @@
+import Component from '@ember/component';
+
+export default Component.extend({
+	classNames: ['img-fancy-load'],
+	placeholderSrc: 'http://placehold.jp/d1d1d1/707070/400x300.png?text=No%20Image',
+	showImg: false,
+
+	didInsertElement() {
+		this._super(...arguments);
+		this._initImage();
+	},
+
+	_initImage() {
+		const img = new Image();
+
+		img.src = this.get('src');
+
+		img.addEventListener('load', () => {
+			// this is to handle if the component has been destroyed before finishing in tests
+			if (this.isDestroyed) {
```

Comment:
Is a little strange that we need this code in only for tests, ideally tests should not require us to write extra code.

It also makes me thing that this is a potential memory leak.

Any other ways we could do this? Maybe remove the listeners when the component destroys?

Reply:
It had to do with something @biancamariacampean suggested. I've refactored it